### PR TITLE
doc/user: fix lingering source/sink `SIZE` issues

### PR DESCRIPTION
**Pull request by benesch**
* Add documentation for `ALTER SINK` (#13889).
  * Correct documentation for Kafka sources to not claim that `3xsmall` is the default.
  * Link to the "sizing a source" section from the individual source type pages.
  * Use the spelling `SIZE = 'size'` consistently throughout the docs, since that appears to be the most common spelling.

### Motivation

  * This PR fixes some bugs reported in Slack.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

File: doc/user/content/sql/create-sink.md [hunk 1]
```diff
@@ -207,6 +198,26 @@ JOIN mz_kafka_sinks USING (id);
...
+#### Sizing a sink
+
+To provision a specific amount of CPU and memory to a sink on creation, use the `SIZE` option:
+
+```sql
+CREATE SINK snk FROM src
+INTO KAFKA CONNECTION kafka_connection (TOPIC 'snk')
+FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_connection
+ENVELOPE DEBEZIUM
+WITH (SIZE = 'xsmall');
```
Comment (morsapaes) [inline doc/user/content/sql/create-sink.md:210]:
```suggestion
CREATE SINK snk FROM src
  INTO KAFKA CONNECTION kafka_connection (TOPIC 'snk')
  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_connection
  ENVELOPE DEBEZIUM
  WITH (SIZE = 'xsmall');
```

**Review by morsapaes — APPROVED** [review]
🙇‍♀️

File: doc/user/content/sql/create-sink.md (hunk 1 above)
Comment (benesch) [reply doc/user/content/sql/create-sink.md:210]: Mind if I leave this one as is for now? It's consistent with all the rest of the `CREATE SINK` examples in this doc AFAICT. No objection to reformatting them all though!

Comment (morsapaes) [reply doc/user/content/sql/create-sink.md:210]: Sure! Was just looking for revenge from yesterday's PR, but then noticed there's no consistency for sinks.

**Pull request merged**